### PR TITLE
Fix `void` calls on iOS

### LINKED ISSUE #191: iOS channel methods blocked
In PR #180 I added a fix for iOS that was reverted in https://github.com/Instabug/Instabug-Flutter/commit/e7bd9ef4e965685cf3777639ff2f0d99cbcb60fc.

I'm not sure why this was done, but it means that if a caller from the Dart code uses `await` then the `Future` will never complete and the app will be blocked.

### PR DESCRIPTION
**Pull request by via-guy**
## Summary of Changes
Fixes #191

As mentioned in the issue, my fix for returning from method calls to iOS was undone for an unknown reason. It blocks returns when calling `await` on `Instabug` calls from iOS. It's a pretty serious issue.

I've also re-enabled the `unawaited_futures` analysis option to show that it works now.

## Checklist
- [x] In addition to adding the new API to Dart, I have also added its implementation to both iOS and Android.
- [x] I have added at least 1 unit test in Dart for the new API and made sure the tests pass locally.
- [x] I have updated [README.md](https://github.com/Instabug/Instabug-Flutter/blob/master/README.md) to add the new API.

### CODE REVIEW

**Comment by via-guy** [comment]
@AliAbdelfattah @ymabdallah can one of you please look into this? It's a serious issue with the code.

File: ios/Classes/InstabugFlutterPlugin.m [hunk 1]
```diff
@@ -20,36 +20,40 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
...
+        const char *type = [signature methodReturnType];
+
+        if (result != nil) {
+            if (strcmp(type, "v") != 0) {
+                void *returnVal;
+                [inv getReturnValue:&returnVal];
+                NSObject *resultSet = (__bridge NSObject *)returnVal;
+                result(resultSet);
+            } else {
+                result(nil);
```
Comment (via-guy) [inline ios/Classes/InstabugFlutterPlugin.m:51]: This is the important fix. If you don't call the completion block then the Dart code will never complete the Future.

**Comment by via-guy** [comment]
@ahmed1hisham @AliAbdelfattah @ymabdallah can someone please review this?

**Comment by AliAbdelfattah** [comment]
@via-guy sorry this's been taking long, we'll try to prioritize this 👌

**Comment by via-guy** [comment]
@AliAbdelfattah added missing CHANGELOG entry. Can you please check it?

**Comment by via-guy** [comment]
@AliAbdelfattah we've been waiting months for this. Please can you look into merging it?

**Review by TheBuggedYRN — CHANGES_REQUESTED** [review]
Sorry for the late response @via-guy. We are working on this now, and it will get merged as soon as we resolve the comments.

**Comment by TheBuggedYRN** [comment]
We need to exclude other changes from 25da1ca7b3cddf1276ed28e747aeeaebfae25ed8, too.

**Review by TheBuggedYRN — CHANGES_REQUESTED** [review]
We need to exclude other changes from https://github.com/Instabug/Instabug-Flutter/commit/25da1ca7b3cddf1276ed28e747aeeaebfae25ed8, too.

**Comment by via-guy** [comment]
@TheBuggedYRN done

**Comment by TheBuggedYRN** [comment]
Please [allow edit from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests), because we need to modify some unit tests related to this issue.

**Comment by via-guy** [comment]
@TheBuggedYRN it's already enabled

**Review by TheBuggedYRN — APPROVED** [review]

**Comment by TheBuggedYRN** [comment]
Thanks @via-guy for your contribution!

**Pull request merged**